src/App.tsx

```diff
@@ -15,71 +16,80 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+type StateType = {
+  selectedGood: string | null,
+};
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export class App extends React.Component {
+  state: StateType = {
+    selectedGood: 'Jam',
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  clearCartHandler = () => {
+    this.setState({ selectedGood: null });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  selectGoodHandler(name: string) {
+    this.setState((prevState: StateType) => ({
+      selectedGood: prevState.selectedGood !== name
+        ? name
+        : null,
+    }));
+  }
 
-        <tr data-cy="Good">
-          <td>
+  render() {
+    return (
+      <main className="section container">
+        {this.state.selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {this.state.selectedGood}
```

Comment:
it would be nice to decompose `this.state` to not it repeat each time

`const { selectedGood } = this.state;`